### ISSUE TITLE
suspicious_to_owned lint for Cow has invalid MaybeIncorrect fix

### ISSUE THREAD

**Issue by DesmondWillowbrook** [rust-lang/rust-clippy#10294]
### Summary

`Applicability::MaybeIncorrect` is meant for fixes which may not be the user's intention, but must result in valid Rust code. However, the current suggestion when applied results in incorrect Rust code.

ref: [`suspicious_to_owned` lint page](https://rust-lang.github.io/rust-clippy/master/#suspicious_to_owned)

### I tried this code:

```rust
let _ = Cow::Borrowed(&()).to_owned();
```

### I expected to see this happen:

I would see two suggested fixes that would replace this by:
```rust
let _ = Cow::Borrowed(&()).clone();
```
or
```rust
let _ = Cow::Borrowed(&()).into_owned();
```

###  Instead, this happened:
Applying the single suggested fix:

```rust
let _ = Cow::Borrowed(&()).clone()` or `Cow::Borrowed(&()).into_owned();
```

### Version

```text
rustc 1.65.0 (897e37553 2022-11-02)
binary: rustc
commit-hash: 897e37553bba8b42751c67658967889d11ecd120
commit-date: 2022-11-02
host: x86_64-unknown-linux-gnu
release: 1.65.0
LLVM version: 15.0.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
